readthedocs/proxito/middleware.py

```diff
@@ -318,18 +318,31 @@ def add_cors_headers(self, request, response):
         accepted by browsers. However, we cannot expose these headers for
         documentation that's not PUBLIC.
         """
+
+        # Disable CORS on "Read the Docs for Business" for now.
+        # We want to be pretty sure this logic is OK before enabling it there.
+        if settings.ALLOW_PRIVATE_REPOS:
+            return
+
         project_slug = getattr(request, "path_project_slug", "")
         version_slug = getattr(request, "path_version_slug", "")
+        host = request.get_host()
```

Comment:
I don't think this will work, it should get the host from the Origin header. get_host() is the current domain being hit, not from where the resource was requested.